app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt

```diff
@@ -122,10 +117,80 @@ class SearchActivity : AppCompatActivity() {
         inputEditText.setText(currentValue)
     }
 
+    private fun clearButtonVisibility(s: CharSequence?): Int {
+        return if (s.isNullOrEmpty()) {
+            View.GONE
+        } else {
+            View.VISIBLE
+        }
+    }
+
+
+    private fun sendRequest() {
+        if (inputEditText.text.isNotEmpty()) {
+            errLayout.visibility = View.GONE
+            iTunesService.findTrack(inputEditText.text.toString())
+                .enqueue(object : Callback<TracksResponse> {
+                    override fun onResponse(
+                        call: Call<TracksResponse>,
+                        response: Response<TracksResponse>
+                    ) {
+                        if (response.code() == 200) {
+                            trackList.clear()
+                            if (response.body()?.results?.isNotEmpty() == true) {
+                                trackList.addAll(response.body()?.results!!)
+                                adapter.notifyDataSetChanged()
+                            }
+                            if (trackList.isEmpty()) {
+                                showMessage(
+                                    getString(R.string.nothing_found),
+                                    R.drawable.icon_nothing_found
+                                )
+                            }
+                        } else {
+                            showMessage(
+                                getString(R.string.something_went_wrong),
+                                R.drawable.icon_network_problem,true
+                            )
+                        }
+                    }
+
+                    override fun onFailure(call: Call<TracksResponse>, t: Throwable) {
+                        showMessage(
+                            getString(R.string.something_went_wrong),
+                            R.drawable.icon_network_problem,true
+                        )
+                    }
+                })
+            true
+        }
+    }
+
+
+    private fun showMessage(text: String, icon: Int, showButton: Boolean=false) {
```

Comment:
```suggestion
    private fun showMessage(text: String, @DrawableRes icon: Int, showButton: Boolean=false) {
```
Аннотация нужна, чтобы в параметр `icon` можно было передавать не любое число, а только айдишку ресурса из `drawables`, это дополнительная защита

Reply:
Текст здесь тоже можно передать с помощью id - `R.string.error_message`, для такого тоже существует аннотация `@StringRes`